### PR TITLE
[#1632] fix(jdbc-mysql): Supporting create not null column for MySQL table.

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add `NOT NULL` key word in create table sentence if column type is NOT NULL.

### Why are the changes needed?

We should support creating a table with not null columns.

Fix: #1632

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

Add some ITs.

### CODE REVIEW

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/trino/TrinoConnectorIT.java [hunk 1]
```diff
@@ -1105,6 +1105,85 @@ void testMySQLTableCreatedByGravitino() throws InterruptedException {
...
+
+    sql =
+        String.format(
+            "show create table \"%s.%s\".%s.%s", metalakeName, catalogName, schemaName, tableName);
+
+    data = containerSuite.getTrinoContainer().executeQuerySQL(sql).get(0).get(0);
+    Assertions.assertTrue(data.contains("age integer NOT NULL"));
+    Assertions.assertTrue(data.contains("address varchar(20) NOT NULL"));
+  }
+
```
Comment (diqiu50) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/trino/TrinoConnectorIT.java:1186]: I suggest you add this test logic to integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-mysql/00002_alter_table.sql

Comment (yuqi1129) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/trino/TrinoConnectorIT.java:1186]: added

**Comment by yuqi1129** [comment]
@diqiu50 
Please take another look.

File: integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-mysql/00002_alter_table.sql [hunk 2]
```diff
@@ -18,7 +18,10 @@ show create table "test.gt_mysql".gt_db1.tb01;
 alter table "test.gt_mysql".gt_db1.tb01 alter column salary set data type bigint;
 show create table "test.gt_mysql".gt_db1.tb01;
```
Comment (diqiu50) [inline integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-mysql/00002_alter_table.sql:20]: We should not remove  the `Alter table add column with null` test case.

Comment (yuqi1129) [reply integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-mysql/00002_alter_table.sql]: Fixed

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java [hunk 3]
```diff
@@ -435,8 +434,7 @@ private String updateColumnCommentFieldDefinition(
     return "MODIFY COLUMN " + col + appendColumnDefinition(updateColumn, new StringBuilder());
   }
 
-  private String addColumnFieldDefinition(
-      TableChange.AddColumn addColumn, CreateTable createTable) {
+  private String addColumnFieldDefinition(TableChange.AddColumn addColumn) {
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java:437]: besides trino test, could you add a mysql test?

**Comment by jerryshao** [comment]
Do we also need to support this for PG?

**Comment by yuqi1129** [comment]
> Do we also need to support this for PG?

No, I have checked PG has no such issue. 
image

image

**Pull request merged**